fix(validation): additional URN validation adjustments (#11973)
datahub-project/datahub: docs/what/urn.md

## Restrictions

There are a few restrictions when creating an URN:

The following characters are not allowed anywhere in the URN

1. Parentheses are reserved characters in URN fields: `(` or `)`

> 2. URN separator UTF-8 character `␟`

Reviewer: ```suggestion
2. The "unit separator" unicode character `␟` (U+241F)
```

The following characters are not allowed within an URN tuple.

1. Commas are reserved characters in URN tuples: `,`

> Example: Adding another `,` between the parentheses in this URN `urn:li:dashboard:(looker,dashboards.thelook)` is invalid.

Reviewer: the phrasing is a bit confusing here 
```suggestion
Example: `urn:li:dashboard:(looker,dashboards.thelook)` is a valid urn, but `urn:li:dashboard:(looker,dashboards.the,look)` is invalid.
```
